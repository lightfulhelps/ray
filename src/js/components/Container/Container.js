// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ChildrenArray<React.Element<any>>,
  className?: string,
  isFluid?: boolean,
  tag?: string,
};

const Container = ({ children, className, isFluid, tag: Tag = 'div', ...other }: Props) => {
  const classes = classNames(className, isFluid ? 'container-fluid' : 'container');

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default Container;
