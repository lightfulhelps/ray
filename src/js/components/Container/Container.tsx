import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ReactNode;
  className?: string;
  isFluid?: boolean;
  tag?: keyof JSX.IntrinsicElements | typeof React.Component;
  [key: string]: any;
};

const Container: React.FC<Props> = ({
  children,
  className,
  isFluid,
  tag: Tag = 'div',
  ...other
}: Props) => {
  const classes = classNames(className, isFluid ? 'container-fluid' : 'container');

  return (
    <Tag {...other} className={classes}>
      {children}
    </Tag>
  );
};

export default Container;
