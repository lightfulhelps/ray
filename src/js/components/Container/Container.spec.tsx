import React from "react";
import { shallow } from "enzyme";
import merge from "lodash/merge";
import Container from "./Container";

const setup = (overrides = {}) => {
  const props = merge({
    children: <p>Children</p>
  }, overrides);
  const wrapper = shallow(<Container {...props} />);

  return {
    wrapper,
    props
  };
};

describe('<Container />', () => {
  it('should render', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the children prop', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.children().text()).toEqual('Children');
  });

  it('should handle the className prop', () => {
    const {
      wrapper
    } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('container')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the isFluid prop', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.hasClass('container')).toBe(true);
    expect(wrapper.hasClass('container-fluid')).toBe(false);

    wrapper.setProps({ isFluid: true });

    expect(wrapper.hasClass('container')).toBe(false);
    expect(wrapper.hasClass('container-fluid')).toBe(true);
  });

  it('should handle the tag prop', () => {
    const {
      wrapper
    } = setup();

    expect(wrapper.type()).toBe('div');

    wrapper.setProps({ tag: 'a' });

    expect(wrapper.type()).toBe('a');
  });
});