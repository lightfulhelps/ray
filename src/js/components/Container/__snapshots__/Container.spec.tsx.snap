// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render 1`] = `
<div
  className="container"
>
  <p>
    Children
  </p>
</div>
`;
